docs/rules/no-untyped-mock-factory.md

```diff
@@ -0,0 +1,57 @@
+# Disallow using `jest.mock()` factories without an explicit type parameter (`no-untyped-mock-factory`)
+
+<!-- end auto-generated rule header -->
+
+By default, `jest.mock` and `jest.doMock` allow any type to be returned by a
+mock factory. A generic type parameter can be used to enforce that the factory
+returns an object with the same shape as the original module, or some other
+strict type. Requiring a type makes it easier to use TypeScript to catch changes
+needed in test mocks when the source module changes.
+
+## Rule details
+
+This rule triggers a warning if `mock()` or `doMock()` is used without a generic
+type parameter or return type.
+
+The following patterns are considered errors:
+
+```ts
+jest.mock('../moduleName', () => {
```

Comment:
nit: these feel a bit cramped to me - it'd be nice to space them out with some newlines between each example (same with the second codeblock below)